modules/internal/ChapelLocale.chpl

```diff
@@ -48,8 +48,29 @@ module ChapelLocale {
     // The parent of the root locale is nil (by definition).
     const parent : locale;
 
-    // To be removed from the required interface once legacy code is adjusted.
-    // Modified in RootLocale.init().
+    pragma "no doc" var nPUsPhysAcc: int;    // HW cores, OS accessible
+    pragma "no doc" var nPUsPhysAll: int;    // HW cores, all
+    pragma "no doc" var nPUsLogAcc: int;     // HW threads, OS accessible
+    pragma "no doc" var nPUsLogAll: int;     // HW threads, all
+
+    /*
```

Comment:
This description could be confusing for users who don't know much about hardware and really just want to know what function tells them how many cores there are. I would imagine most people will know about physical cores vs logical/hyperthreaded cores, but will be unaware of os limitations.

Might be nice to start with defining PU (I'm assuming processing unit?). It might also be nice to put in words what the default will be. i.e. that `numPUs()` will return the number of accessible physical cores so they don't have to figure it out based on the default values.